chatbots-basecamp-lambda/src/main/java/io/micronaut/chatbots/basecamp/lambda/Handler.java

```diff
@@ -68,8 +72,19 @@ public Handler(ApplicationContextBuilder applicationContextBuilder) {
     @NonNull
     protected boolean validate(@NonNull APIGatewayProxyRequestEvent request) {
         return parseHeader(request, HttpHeaders.USER_AGENT)
-            .map(value -> value.equalsIgnoreCase("Basecamp 3 Integration Command"))
-            .orElse(false);
+            .map(value -> {
+                boolean result = value.contains("Basecamp");
+                if (result) {
+                    LOG.trace("HTTP Header {}: {}", HttpHeaders.USER_AGENT, value);
```

Comment:
Don't we need to wrap this in a check that trace is enabled?

Reply:
And then maybe extracting a method like:

```
    private static boolean isBasecamp(String value) {
        boolean result = value.contains("Basecamp");
        if (result) {
            if (LOG.isTraceEnabled()) {
                LOG.trace("HTTP Header {}: {}", HttpHeaders.USER_AGENT, value);
            }
        } else {
            if (LOG.isWarnEnabled()) {
                LOG.warn("Rejecting request because HTTP Header {}: {} does not contain the word Basecamp", HttpHeaders.USER_AGENT, value);
            }
        }
        return result;
    }
```

And using

```
    @Override
    @NonNull
    protected boolean validate(@NonNull APIGatewayProxyRequestEvent request) {
        return parseHeader(request, HttpHeaders.USER_AGENT)
            .map(Handler::isBasecamp)
            .orElseGet(() -> {
                LOG.warn("Rejecting request because HTTP Header {} not present", HttpHeaders.USER_AGENT);
                return false;
            });
    }
```

Makes it more readable?

Reply:
good idea https://github.com/micronaut-projects/micronaut-chatbots/pull/59/commits/22ba5ea9ec434a9f2e62e24856e529ddf88b1406